File: rperz/QSLib/QSLib/Expressions/Binary/Multiply.cs
using QSLib.Types;
using QSLib.Values;
using System.ComponentModel;

namespace QSLib.Expressions.Binary
{
    public class Multiply : Binary_Expression 
    {
        public Multiply(QSExpression a, QSExpression b, int linenr)
            : base(a, b, linenr)
        {
            base._operator = "*";
        }

        public override Value Evaluate()
        {
            this._value = this._left.Evaluate().Multiply(this._right.Evaluate());
            this.OnPropertyChanged("GetValue");
            return this._value;
        }


        public override QSType Type
        {
            get { return new NumberType(); }
        }
    }
}

File: tvkanters/PolyQL/src/nl/uva/polyql/model/expressions/operators/Multiply.java
package nl.uva.polyql.model.expressions.operators;

import nl.uva.polyql.model.expressions.operations.Operator;
import nl.uva.polyql.model.values.NumberValue;
import nl.uva.polyql.model.values.Value;

public class Multiply extends Operator {

    public static final String SYNTAX = "*";

    @Override
    public String getSyntax() {
        return SYNTAX;
    }

    @Override
    public Value<?> performOperation(final NumberValue left, final NumberValue right) {
        return new NumberValue(left.getValue() * right.getValue());
    }

}
